### PR TITLE
Rclone storage backend

### PR DESCRIPTION
**Pull request by dalbani**
# What problem are we solving?

SeaweedFS currently only supports S3 as storage backend for the Cloud tier functionality: https://github.com/seaweedfs/seaweedfs/wiki/Cloud-Tier.  
As a user of SeaweedFS, I'd like to be able to have more choice of storage protocols and providers. 

# How are we solving the problem?

This PR adds support for Rclone, well known for its versatility, as storage backend for Cloud Tier.

# Status / feedback

First, I have very little experience with programming in Go, so feedback is welcome.
Second, there are a couple of areas that probably need work and further improvement:
1. ~~Implement progress for `{download,upload}ViaRclone` functions.~~ ✅
2. ~~Add logging for Rclone related operations~~ ✅
    - Setting the `RCLONE_LOG_LEVEL=DEBUG` environment variable makes the Rclone library be more verbose. 

That being said, I've been using this functionality for weeks and have uploaded several terabytes of data via this Rclone backend (to OneDrive) without any error.

# How is the PR tested?

Manually (so far):
1. Configured a remote in Rclone, e.g. here called `onedrive-remote`.
   ```
   [onedrive-remote]
   type = onedrive
   drive_id = 123456789
   drive_type = personal
   ...
   ```
2. Configured SeaweedFS to use this Rclone remote:
   ```
   [storage.backend]
   [storage.backend.rclone.default]
   enabled = true
   remote_name = "onedrive-remote"
   ```
3. Filled SeaweedFS volumes with data (via the S3 API).
4. Triggered tiering in `weed shell` with:
   ```
   volume.tier.upload -dest rclone
   ```
5. Checked that the files were present in the storage provider (OneDrive in this case) configured in Rclone.
6. Retrieved some files via the S3 API and confirmed that they were consistent with what I originally put in.

# Checks
- [ ] I have added unit tests if possible.
- [ ] I will add related wiki document changes and link to this PR after merging.

### CODE REVIEW

**Comment by chrislusf** [comment]
Very nice! Please consider adding related wiki pages when this is merged.

**Comment by dalbani** [comment]
Okay, let's say the code itself is ready for review. Feedback is welcome!

File: weed/storage/backend/rclone_backend/rclone_backend.go [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+	remoteName string
+	fs         fs.Fs
+}
+
+func newRcloneBackendStorage(configuration backend.StringProperties, configPrefix string, id string) (s *RcloneBackendStorage, err error) {
+	s = &RcloneBackendStorage{}
+	s.id = id
+	s.remoteName = configuration.GetString(configPrefix + "remote_name")
+
+	configfile.Install()
```
Comment (chrislusf) [inline weed/storage/backend/rclone_backend/rclone_backend.go]: Should this line only be called once?

**Review by chrislusf — APPROVED** [review]

File: weed/storage/backend/rclone_backend/rclone_backend.go (hunk 1 above)
Comment (dalbani) [reply weed/storage/backend/rclone_backend/rclone_backend.go]: Good suggestion, this also crossed my mind.
I moved it into the `init()` method.

**Comment by dalbani** [comment]
>Very nice! Please consider adding related wiki pages when this is merged.

It was *long* overdue but I've added some documentation about this feature on https://github.com/seaweedfs/seaweedfs/wiki/Cloud-Tier#rclone.

**Comment by chrislusf** [comment]
as mentioned in https://github.com/seaweedfs/seaweedfs/issues/5044
the binary size reduced from 95,571,392 bytes to 65,955,760 bytes with https://github.com/seaweedfs/seaweedfs/commit/7a50762b9ed05f1410a24f52571b49a25cd561eb

**Pull request merged**